File: src/leetcode/2026/06/Solution1021.cpp
#include <string>
#include <iostream>
using namespace std;

class Solution1021 {
public:
    string removeOuterParentheses(string s) {
        string result;
        int openCount = 0;

        for (char c : s) {
            if (c == '(') {
                if (openCount > 0) {
                    result += c;
                }
                openCount++;
            } else {
                openCount--;
                if (openCount > 0) {
                    result += c;
                }
            }
        }
        return result;
    }
};

int main () {
    Solution1021 solution;

    string s1 = "(()())(())";
    string s2 = "(()())(())(()(()))";
    string s3 = "()()";
    cout << solution.removeOuterParentheses(s1) << endl; // "()()()"
    cout << solution.removeOuterParentheses(s2) << endl; // "()()()()(())"
    cout << solution.removeOuterParentheses(s3) << endl; // ""

    return 0;
}

File: src/leetcode/2026/06/Solution1021.java
public class Solution1021 {
    public String removeOuterParentheses(String s) {
        StringBuilder result = new StringBuilder();
        int openCount = 0;

        for (char c : s.toCharArray()) {
            if (c == '(') {
                if (openCount > 0) {
                    result.append(c);
                }
                openCount++;
            } else {
                openCount--;
                if (openCount > 0) {
                    result.append(c);
                }
            }
        }

        return result.toString();
    }

    public static void main(String[] args) {
        Solution1021 solution = new Solution1021();

        String s1 = "(()())(())";
        System.out.println(solution.removeOuterParentheses(s1)); // Output: "()()()"

        String s2 = "(()())(())(()(()))";
        System.out.println(solution.removeOuterParentheses(s2)); // Output: "()()()()(())"

        String s3 = "()()";
        System.out.println(solution.removeOuterParentheses(s3)); // Output: ""
    }
}
